components/infobox/wikis/mobilelegends/infobox_hero_custom.lua

```diff
@@ -0,0 +1,180 @@
+---
+-- @Liquipedia
+-- wiki=mobilelegends
+-- page=Module:Infobox/Unit/Hero
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Unit = require('Module:Infobox/Unit')
+local String = require('Module:StringUtils')
+local Namespace = require('Module:Namespace')
+local Template = require('Module:Template')
+local Math = require('Module:Math')
+local HeroWL = require('Module:HeroWL')
+local Class = require('Module:Class')
+
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Center = require('Module:Infobox/Widget/Center')
+local Title = require('Module:Infobox/Widget/Title')
+local Header = require('Module:Infobox/Widget/Header')
+local Breakdown = require('Module:Infobox/Widget/Breakdown')
+
+local CustomHero = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+
+local _pagename = mw.title.getCurrentTitle().text
+local _frame
+
+local _BATTLE_POINTS_ICON = '[[File:Mobile_Legends_BP_icon.png|x16px|Battle Points|link=Battle Points]]'
+local _DIAMONDS_ICON = '[[File:Mobile_Legends_Diamond_icon.png|Diamonds|x16px|link=Diamonds]]'
+
+function CustomHero.run(frame)
+	local unit = Unit(frame)
+	_args = unit.args
+	_frame = frame
+	_args.informationType = 'Hero'
+
+	unit.getWikiCategories = CustomHero.getWikiCategories
+	unit.setLpdbData = CustomHero.setLpdbData
+	unit.createWidgetInjector = CustomHero.createWidgetInjector
+
+	return unit:createInfobox(frame)
+end
+
+function CustomInjector:addCustomCells()
+	local widgets = {
+		Cell{name = 'Resource Bar', content = {_args.secondarybar}},
+		Cell{name = 'Secondary Bar', content = {_args.secondarybar1}},
+		Cell{name = 'Secondary Attributes', content = {_args.secondaryattributes1}},
+		Cell{name = 'Release Date', content = {_args.releasedate}},
+		Cell{name = 'Species', content = {_args.species}},
+		Cell{name = 'Year of birth', content = {_args.birth}},
+		Cell{name = 'Faction(s)', content = {_args.factions}},
+	}
+
+	if not (
+		String.isEmpty(_args.hp) and
+		String.isEmpty(_args.hplvl) and
+		String.isEmpty(_args.hpreg) and
+		String.isEmpty(_args.hpreglvl)
+	) then
```

Comment:
Same comment as in infobox item

Reply:
Also, is it really correct?

Reply:
i doubt that that logic is good for the long long list of cells that follows
i will go through the infobox modules on that wiki and add a few changes
since this PR is from a fork hesketh will have to update the PRs tomorrow/monday^^

Reply:
Merged into https://github.com/Liquipedia/Lua-Modules/pull/714#discussion_r794423716